plugins/Chart/__init__.py

```diff
@@ -1,48 +1,76 @@
 from discord.ext import commands
 from Plugin import AutomataPlugin
 import discord
-import matplotlib.pyplot as plt
 import io 
 from typing import List
+import requests
+import datetime
 
-def parseArguments(args: str) -> List[float]:
+# ==== UTIL FUNCTIONS =====
+# TODO: Write unit tests
+
+def parseArguments(args: str) -> List[str]:
     """ Takes the arguments from barplot function 
-        And converts it to a list of usable numbers. """
+        And converts it to a list of string numbers. """
     
     if "," in args:
         temp = args.strip().split(",")
         return 
     else:
         temp =  args.strip().split()
     
-    return list(map(float, temp))
+    # Temp to ensure that all parameters are numbers
+    temp = list(map(float, temp))
+    return list(map(str, temp))
+
+def getErrorEmbed(title="Error Title", message="Error Message") -> discord.Embed:
+    """
+    Generate an embed for error messages
+    """
+    emb = discord.Embed(
+        title=message,
+        timestamp=datetime.datetime.now(),
+        colour = discord.Colour(0xEF440E)
+    )
+    emb.set_author(
+        name=title,
+        icon_url="https://i.imgur.com/UjdPxZw.png" # Potentially is not the safest. TODO: Find alternative solution
+    )
+
+    return emb
 
 class Chart(AutomataPlugin):
     """ A discord plugin for some chart related commands """
 
     @commands.command()
     async def barplot(self, ctx: commands.Context, *, args: str):
-        """ Plots a bar chart """
+        """ 
+        Plots a bar chart 
+        Fetch the barplot from an external API instead of using matplotlib internally
+        """
+        ENDPOINT = "https://funapi.onrender.com/api/barplot/"
```

Comment:
Putting this const at the top beneath the imports would be nice, and renaming it to what the endpoint is for.